x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_style_property.tsx

```diff
@@ -345,4 +355,44 @@ export class DynamicStyleProperty<T>
       />
     );
   }
+
+  getMbPropertyName() {
+    if (!this._field) {
+      return '';
+    }
+
+    let targetName;
+    if (this.supportsMbFeatureState()) {
+      // Base case for any properties that can support feature-state (e.g. color, size, ...)
+      // They just re-use the original property-name
+      targetName = this._field.getName();
+    } else {
+      if (this._field.canReadFromGeoJson()) {
+        // Geojson-sources can support rewrite
+        // e.g. field-formatters will create duplicate field
+        targetName = this._field.supportsAutoDomain()
+          ? getComputedFieldName(this.getStyleName(), this._field.getName())
+          : this._field.getName();
+      } else {
+        // Non-geojson sources (e.g. mvt)
+        targetName = this._field.getName();
+      }
+    }
+    return targetName;
+  }
+
+  getMbPropertyValue(
+    rawValue: string | number | null | undefined
+  ): string | number | null | undefined {
+    return this.supportsMbFeatureState() ? getNumericalMbFeatureStateValue(rawValue) : rawValue;
```

Comment:
Should this be based on whether the property is ordinal or not instead of supportsMbFeatureState? So how about this instead
```
return this.isOrdinal() ? getNumericalMbFeatureStateValue(rawValue) : rawValue;
```

Reply:
This would end up slightly altering the previous logic. It would end up being equivalent (since feature-state only supports numerical values).

Should I change?

Reply:
added comment